nomad/vault.go

```diff
@@ -516,7 +566,28 @@ func (v *vaultClient) parseSelfToken() error {
 		}
 	}
 
+	// Store the token data
+	data.Root = root
+	v.tokenData = &data
+
+	// The criteria that must be met for the token to be valid are as follows:
+	// 1) If token is non-root or is but has a creation ttl
```

Comment:
We don't restrict people to that. That is an organizational decision not one Nomad will enforce on people